src/apps/content-editor/src/app/components/Editor/Field/Field.js

```diff
@@ -529,22 +550,30 @@ export default connect(state => {
           props.allFields,
           props.allItems,
           relatedFieldZUID,
-          relatedModelZUID
+          relatedModelZUID,
+          langID,
+          value
         );
       }, [
-        Object.keys(props.allFields).length,
```

Comment:
I am concerned this change is a performance regression. The reason we only ran the memo on length change of `allFields` and `allItems` is because a change to any property on those objects will trigger this. Which in turn can trigger renders across other components in the view. With highly related content items this becomes a significant rendering performance problem.